index.js

```diff
@@ -131,6 +132,26 @@ function compression (options) {
       return this
     }
 
+    res.removeListener = function removeListener (type, listener) {
+      if (!listeners || type !== 'drain') {
+        return _removeListener.call(this, type, listener)
+      }
+
+      if (stream) {
+        return stream.removeListener(type, listener)
+      }
+
+      // remove buffered listener
+      for (var i = 0; i < listeners.length; i++) {
+        if (listeners[i][0] === type && listeners[i][1] === listener) {
+          listeners.splice(i, 1)
+          break
```

Comment:
I think we cannot actually `break` here; at least with the current queue implementation. If you `res.on('drain', fn).on('drain', fn)`, `fn` ends up in the array twice, for example, and this will only remove the first. We should add a test for this case to the test suite too.